IT unstable test `TestLoadData`

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
in ci https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_coverage/detail/tidb_ghpr_coverage/1387/pipeline
```bash 
[2021-11-29T09:17:27.003Z] --- FAIL: TestLoadData (1.30s)

[2021-11-29T09:17:27.003Z] panic: sync: WaitGroup is reused before previous Wait has returned [recovered]

[2021-11-29T09:17:27.003Z] 	panic: sync: WaitGroup is reused before previous Wait has returned

[2021-11-29T09:17:27.003Z] 

[2021-11-29T09:17:27.003Z] goroutine 6118 [running]:

[2021-11-29T09:17:27.003Z] testing.tRunner.func1.2(0x3953520, 0x439cfd0)

[2021-11-29T09:17:27.003Z] 	/usr/local/go/src/testing/testing.go:1143 +0x332

[2021-11-29T09:17:27.003Z] testing.tRunner.func1(0xc000f0b080)

[2021-11-29T09:17:27.003Z] 	/usr/local/go/src/testing/testing.go:1146 +0x4b6

[2021-11-29T09:17:27.003Z] panic(0x3953520, 0x439cfd0)

[2021-11-29T09:17:27.004Z] 	/usr/local/go/src/runtime/panic.go:965 +0x1b9

[2021-11-29T09:17:27.004Z] sync.(*WaitGroup).Wait(0xc0108fe230)

[2021-11-29T09:17:27.004Z] 	/usr/local/go/src/sync/waitgroup.go:132 +0xae

[2021-11-29T09:17:27.004Z] github.com/tikv/client-go/v2/tikv.(*KVStore).Close(0xc0108fe120, 0xc000f0b080, 0xc011239ec8)

[2021-11-29T09:17:27.004Z] 	/nfs/cache/mod/github.com/tikv/client-go/v2@v2.0.0-alpha.0.20211118154139-b11da6307c6f/tikv/kv.go:329 +0x47

[2021-11-29T09:17:27.004Z] github.com/pingcap/tidb/store/mockstore/mockstorage.(*mockStorage).Close(0xc02d512600, 0x3fe3700, 0x3eeef07)

[2021-11-29T09:17:27.004Z] 	/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/store/mockstore/mockstorage/storage.go:117 +0x45

[2021-11-29T09:17:27.004Z] github.com/pingcap/tidb/server.createTidbTestSuite.func2()

[2021-11-29T09:17:27.004Z] 	/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/server/tidb_test.go:100 +0x76

[2021-11-29T09:17:27.004Z] github.com/pingcap/tidb/server.TestLoadData(0xc000f0b080)

[2021-11-29T09:17:27.004Z] 	/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/server/tidb_serial_test.go:47 +0xd8

[2021-11-29T09:17:27.004Z] testing.tRunner(0xc000f0b080, 0x3fe3a50)

[2021-11-29T09:17:27.004Z] 	/usr/local/go/src/testing/testing.go:1193 +0xef

[2021-11-29T09:17:27.004Z] created by testing.(*T).Run

[2021-11-29T09:17:27.004Z] 	/usr/local/go/src/testing/testing.go:1238 +0x2b3

[2021-11-29T09:17:27.004Z] FAIL	github.com/pingcap/tidb/server	13.248s
```
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/server/tidb_test.go
+++ b/server/tidb_test.go
@@ -95,14 +95,14 @@ func createTidbTestSuite(t *testing.T) (*tidbTestSuite, func()) {
 	ts.waitUntilServerOnline()
 
 	cleanup := func() {
-		if ts.store != nil {
-			ts.store.Close()
-		}
 		if ts.domain != nil {
 			ts.domain.Close()
 		}
 		if ts.server != nil {
 			ts.server.Close()
+		}
+		if ts.store != nil {
+			require.NoError(t, ts.store.Close())
 		}
 	}
 

